### ISSUE TITLE
Copy/Cut/Paste not available in context-menu

### ISSUE THREAD

**Issue by yelvert** [codemirror/CodeMirror#3238]
This is Chrome on OSX 
![image](https://cloud.githubusercontent.com/assets/7054/7483790/8c487f08-f347-11e4-87c9-779048cdf338.png) copy/cut/paste isn't available on any browser on any OS. Im using angular ui-codemirror but ive tried initializing it manually and have the same problem so I don't think its related.

The options I'm using are:
```
$scope.editorOptions = {
              tabSize: 4,
              mode: 'none',
              viewportMargin: Infinity,
              lineWrapping: true,
              extraKeys: {
                'Shift-2': function (cm) {
                  // custom stuff :)
                }
              }
            };
```

**Comment by marijnh** [comment]
Does the problem occur on the demos on http://codemirror.net? I just tried (Chrome 42, Mountain Lion), and copy/paste from the context menu worked fine.

**Comment by yelvert** [comment]
Well, yes, the demos work, and after some digging I realized that the fix I did here https://github.com/yelvert/CodeMirror/commit/714cf11e6cc7d2aea96f51b6dbb0d9ccf178f689 to fix ![image](https://cloud.githubusercontent.com/assets/7054/7497332/5538bef4-f3df-11e4-9e77-fc771341f442.png) is what caused the context menu yo stop working correctly. So my real issue is the text box showing up when right clicking and never going away. If I need to create a new ticket I will.

**Comment by Veejay** [comment]
I'm experiencing the same issue. The demos on CodeMirror's website are fine (I have copy/cut/paste in the context menu), but in the editor we use in our application, no dice in any browser. Is there a workaround I could use to work around this issue? Thank you.

**Comment by marijnh** [comment]
Try to narrow it down -- something in your site is causing it, so remove pieces one by one until it goes away, then figure out what precise line did it. It is likely a CSS conflict, but might be something else.

**Comment by roelvanduijnhoven** [comment]
I can reproduce this in Chrome using the example given in #2154 (http://output.jsbin.com/ihunin/328/). Thus that is not only a FF bug.

The editor is in my case in a modal screen which has transform CSS options. When I move the editor out of the modal screen it *does* work.

Any clue? I guess using a CSS transform should not remove copy/paste functionallity.

**Comment by marijnh** [comment]
In the JSBin case, it seems to be caused by a mysterious no-op CSS transform they inject:

```
body { transform: translate3d(0,0,0); }
```

Why a zero-translate would have an effect on the position of the textarea I don't know.

**Comment by marijnh** [comment]
Actually, some further testing suggests that a 3d transform prevents the trick where CodeMirror puts a textarea under the cursor from working because it makes it appear on the screen asynchronously, possibly due to some interaction with the graphics hardware being needed, which prevents the context menu click from taking effect on that textarea. 

So yeah, 3d transforms on the body break CodeMirror's textarea hack.

**Comment by marijnh** [comment]
Hold on, I misdiagnosed that. It's not about async/sync relayout, it's about a bogus position computation for a `position: fixed` element in a body that has a transform. You can see the effect using this test HTML:

```html
<!doctype html>

<body style="margin: 10px">
  <div style="position: fixed; left: 0px; top: 0px; border: 1px solid black">Fixed at 0,0px</div>
  <button onclick="document.body.style.transform = document.body.style.transform ? '' : 'translate(0,0)'">Toggle transform</button>
</body>
```

**Comment by marijnh** [comment]
Okay, I'll stop spamming soon. It [appears](http://stackoverflow.com/questions/15194313/webkit-css-transform3d-position-fixed-issue) that you can't use `position: fixed` in its normal way inside of a transformed container. This means that JSBin appears to be breaking fixed positioning, and that you have to be pretty careful with transforms in general.

**Comment by roelvanduijnhoven** [comment]
Mm that sucks :/. Thanks for helping me out here!

The thing is: my use case is pretty normal (I guess) as I am rendering Codemirror inside of a Bootstrap modal. The `modal-dialog` class uses a `transform` of `translate3d(0, 0, 0)` (new versions use a 2d transform; but they fail as well).

**Comment by marijnh** [comment]
I think attached patch helps here.

**Comment by roelvanduijnhoven** [comment]
And.. it's gone! That totally fixes the problem. Thank you very much for helping out. Once new release lands I'll update.

PS. also thank you for your awesome book that I'll pass around to new faces at our company ;)

**Comment by ryandemmer** [comment]
I have had to revert this fix - https://github.com/codemirror/CodeMirror/commit/dfeafe0c14478404288dc1f188a0b54baf0f40d5 - when loading CodeMirror inside an iframe.

Does anyone else experience this?
